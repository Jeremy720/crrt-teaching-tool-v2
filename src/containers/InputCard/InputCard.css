.InputCard-text {
  display: flex;
  flex-direction: column;
}

.header-info-container-number {
  display: flex;
  align-items: center;
  height: 2.25rem;
}

.input-text-header {
  text-align: left;
}

.input-number {
  width: 50%;
  height: 1.25rem;
  border-radius: 5px;
  border: 1px solid #a09e9e;
}


.input-container {
  background-color: #fff;
  display: flex;
  flex-direction: column;
  height: 2rem;
  margin: 1rem;
}

.input-error-container {
  width: 50%;
  height: 3rem;
  overflow: scroll;
  width: 80%;
}

.error-add-border {
  border-radius: 5px;
  box-shadow: 1px 1px 1px #d0d6d3;
}

.input-error-text {
  color: red;
  font-size: 0.75rem;
  text-align: left;
  margin: 0.5rem 0 0 0;
  overflow-wrap: normal;
}