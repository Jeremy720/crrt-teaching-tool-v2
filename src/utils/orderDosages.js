const orderDosages = {
  requiredRanges: {
    "sodium": {
      min: 130,
      max: 150
    },
    "potassium": {
      max: 4
    },
    "chloride": {
      min: 95,
      max: 110
    },
    "bicarbonate": {
      min: 20,
      max: 40
    },
    "calcium": {
      min: 0,
      max: 3
    },
    "magnesium": {
      min: 0,
      max: 2
    },
    "phosphorous": {
      min: 0,
      max: 1
    },
    "grossUltraFiltration": {
      min: 0,
      max: 2000
    },
    "replacementFluidFlowRate": {
      min: 0,
      max: 8
    },
  },
  errorMessages: {
    "sodium": "The hospital pharmacy can only compound sodium between 130 and 150.  Use D5W or 3% normal saline if needed",
    "potassium": "The hospital pharmacy will not compound fluid with a potassium above 4 mg/dL",
    "chloride": "The hospital pharmacy can only compound chloride between 95 and 110",
    "bicarbonate": "The hospital pharmacy can only compound bicarbonate between 20 and 40",
    "calcium": "The hospital pharmacy can only compound calcium between 0 and 3 mEq/L",
    "magnesium": "The hospital pharmacy can only compound magnesium between 0 and 2 mgd/L",
    "phosphorous": "The hospital pharmacy cannot compound fluid with > 1mg/dL of phosphorous, due to precipitation of calcium phosphate.",
    "grossUltraFiltration": "The CRRT machine will not accept ultrafiltration rates above 2,000 mL/hour",
    "replacementFluidFlowRate": "The CRRT machine cannot deliver fluid above 8 L/hr",
  }
}

export default orderDosages;