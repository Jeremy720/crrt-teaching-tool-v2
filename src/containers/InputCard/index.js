import React, { Component } from 'react';
import './InputCard.css';
import orderDosages from '../../utils/orderDosages.js';

class InputCard extends Component {
	constructor(props) {
		super(props);
		this.state = {

		}
	}

	render() {
		const { type, enteredInput, handleNumberChange, medicine, dosageErrors } = this.props;
		const { errorMessages, dosageNames } = orderDosages;

		return (
				<article className='input-container'>
					{ type === 'text' &&
							<div className='InputCard-text'>
								<div className='header-info-container'>
									<h4>{ dosageNames[medicine] }</h4>
									<a 
										href='https://github.com/raualex/crrt-teaching-tool-v2' 
										className='textbook-link'
									>
										<i className='far fa-question-circle'></i>
									</a>
								</div>

								<input 
									type='text'
									name={medicine}
									value={isNaN(enteredInput) ? 0 : enteredInput}
									onChange={event => handleNumberChange(event)}
								/>
								<div className='input-error-container'>
									<p className='input-error-text'>{
										dosageErrors.includes(medicine) ? errorMessages[medicine] : ''
									}
									</p>
								</div>
							</div>
					}

					{
						type === 'radio' &&
							<div className='modality-radio'>
								<label>
									<input 
										type='radio'
										name='modality'
										value='Pre-filter CVVH'
										checked={modality === 'Pre-filter CVVH'}
										onChange={this.handleStringChange}
									/>
									Pre-filter CVVH
										<a 
										href='https://github.com/raualex/crrt-teaching-tool-v2' 
										className='textbook-link'
										>
											<i className='far fa-question-circle'></i>
										</a>
								</label>
							</div>						
					}
				</article>
		)
	}
}

export default InputCard;