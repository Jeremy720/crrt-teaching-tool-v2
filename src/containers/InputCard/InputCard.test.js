import React from 'react';
import InputCard from './';
import { shallow } from 'enzyme';
import orderDosages from '../../utils/orderDosages.js';

describe('InputCard', () => {
  let wrapper;
  let mockState;

  beforeEach(() => {
    mockState = {
			modality: 'Pre-filter CVVH',
			sodium: 4,
			potassium: 3,
			chloride: 5,
			bicarbonate: 6,
			calcium: 8,
			magnesium: 9,
			phosphorous : 3,
			grossUltraFiltration: 2,
			bloodFlowRate: 2,
			replacementFluidFlowRate: 9,
			saline3Percent: false,
			d5W: false,
			sodiumPhosphate15mmol100ml: false,
			anticoagulation: 'None',
			readyForSubmission: false,
			dosageErrors: ['sodium']
		}

    wrapper = shallow(<InputCard
								type={'text'}
								radioButtonCategory={null}
								currentInput={'sodium'}
								handleInputChange={jest.fn()}
								dosage={'sodium'} 
								dosageErrors={mockState.dosageErrors}
								key={'sodium'}
					/>);
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});