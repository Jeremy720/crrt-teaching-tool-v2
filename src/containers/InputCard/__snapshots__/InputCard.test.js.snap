// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCard Should render like snapshot 1`] = `ShallowWrapper {}`;
